specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedInstanceDTCs.json

```diff
@@ -0,0 +1,305 @@
+      "get": {
+        "tags": [
+          "ManagedInstanceDTCs"
+        ],
+        "description": "Gets a list of the managed instance DTC settings.",
+        "operationId": "ManagedInstanceDTCs_ListByManagedInstance",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "name": "managedInstanceName",
+            "in": "path",
+            "description": "The name of the managed instance.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the specified managed instance DTC settings.",
+            "schema": {
+              "$ref": "#/definitions/ManagedInstanceDTCListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 ManagementServiceManagedInstanceDtcFeatureDisabled - User tried to interact with Azure SQL Managed Instance DTC settings but DTC feature is currently not supported.\n\n * 400 InvalidDtcVirtualClusterVersion - Virtual cluster version of the Azure SQL Managed Instance \"{0}\" is not supported.\n\n * 404 SubscriptionNotFound - The requested subscription was not found.\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets a list of the managed instance DTC settings.": {
+            "$ref": "./examples/ManagedInstanceDtcList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/dtc/{dtcName}": {
+      "get": {
+        "tags": [
+          "ManagedInstanceDTCs"
+        ],
+        "description": "Gets the managed instance DTC settings.",
+        "operationId": "ManagedInstanceDTCs_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "name": "managedInstanceName",
+            "in": "path",
+            "description": "The name of the managed instance.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "dtcName",
+            "in": "path",
+            "description": "Name of the DTC instance used.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "current"
+            ],
+            "x-ms-enum": {
+              "name": "DtcName",
```

Comment:
>DtcName [](http://example.com/codeflow?start=23&length=7)

The path param cannot be an enum .. Any reason this needs to be modelled like this ?

Reply:
This parameter can only have the value "current" because the DTC resource is a singleton. So the uri goes like: ...managedInstances/{managedInstanceName}/dtc/current. From what I've seen on other SQL API specifications that deal with singletons the parameters are defined like this (some of them use "default" instead of "current"). For example ManagedInstanceEncryptionProtectors in v2020-08-01-preview, or DatabaseSecurityAlertPolicies in v2020-02-02-preview

Reply:
Also, from the ARM wiki: 
"**String types that should have enums**
If a property has a finite set of values (enforced by the service), the string property should be declared as an enum type in the Swagger (using the x-ms-enum annotation) to enable client-side input validation and discovery."

Reply:
Yeah that makes sense. Sry,  I didnt consider singleton objects.